hc_core/src/source_chain/memory/mod.rs

```diff
@@ -36,6 +36,25 @@ impl super::SourceChain for SourceChain {
     // @TODO - appending pairs should fail if hashes do not line up
     // @see https://github.com/holochain/holochain-rust/issues/31
     fn push(&mut self, pair: &super::Pair) {
+        let previous_pair = match pair.header.previous() {
+            Some(h) => self.get(h),
+            None => None,
+        };
```

Comment:
Please take a look at `Option::and_then`. You could write the same in one line:
```rust
let previous_pair = pair.header.previous().and_then(|h| self.get(h));
```

Reply:
thanks, this is exactly what i was reaching for